pkg/validation/validate.go

```diff
@@ -112,6 +112,20 @@ func ValidateLabels(limits LabelValidationLimits, tenantID string, ls []*typesv1
 	if len(ls) == 0 {
 		return NewErrorf(MissingLabels, MissingLabelsErrorMsg)
 	}
+
+	sanitizedNames := make(map[string]bool)
```

Comment:
This is an allocation in the hot path for any profile if needing sanitation or not. Also in the worst case you are adding at most two elements, if I read the code correctly.

Reply:
N/A anymore